yolort/runtime/trt_helper.py

```diff
@@ -89,6 +88,7 @@ def export_tensorrt_engine(
     engine_builder.create_engine(engine_path)
 
 
+@_dependency.requires_module("tensorrt")
```

Comment:
Hi @mattpopovich , could we move this decorator to the following line which we actually call the TensorRT, I guess this will make this dependency more general?

https://github.com/zhiqwang/yolov5-rt-stack/blob/9a27b916da21a0143eb5497ee7ec6dced69f2358/yolort/runtime/y_tensorrt.py#L106

Reply:
We do use TensorRT in this class a few lines below though, so don't we need `@_dependency.requires_module("tensorrt")` where it is? 
https://github.com/zhiqwang/yolov5-rt-stack/blob/cc1ff8bdf6831f0f58d96f66641ebc64a5a8fcb8/yolort/runtime/trt_helper.py#L126-L132

Reply:
And for `y_tensorrt.py`, I can move the generator instead of over the whole `PredictorTRT` class, I can move it to `_build_engine()` but I think I should also place one before `_set_context()` as it also uses TRT:
https://github.com/zhiqwang/yolov5-rt-stack/blob/cc1ff8bdf6831f0f58d96f66641ebc64a5a8fcb8/yolort/runtime/y_tensorrt.py#L120-L123
Thoughts?

Reply:
Seems that the two `_build_engine()` and `_set_context()` must be used together, I personally think it would be sufficient if we put this decorator in the position where TensorRT is actually called at first time. (But I feel it is okay to place both.)

Reply:
> And for y_tensorrt.py, I can move the generator instead of over the whole PredictorTRT class, I can move it to _build_engine() but I think I should also place one before _set_context() as it also uses TRT.

I'm not sure if it's possible to put the decorator inside the function, but we could try putting the decorator in front of the `__init__()`:

https://github.com/zhiqwang/yolov5-rt-stack/blob/9a27b916da21a0143eb5497ee7ec6dced69f2358/yolort/runtime/trt_helper.py#L97-L108

Reply:
I have moved the requested `@_dependency.requires_module()`s. CI/CD is running... Let me know if there's anything else you want changed!

Our of curiosity, why prefer to put the decorator in front of `__init__` instead of `Class EngineBuilder:`?

Reply:
> Our of curiosity, why prefer to put the decorator in front of __init__ instead of Class EngineBuilder:?

Hi @mattpopovich , that's not important, the only reason is to consider whether the Class `EngineBuilder` can be made more applicable/general in the future, if it is put at the top, then there may be problems in defining this class, if it is put in front of `__init__`, at least we can define the class.